src/java/htsjdk/samtools/BamIndexValidator.java

```diff
@@ -84,4 +85,70 @@ public static int exhaustivelyTestIndex(final SamReader reader) { // throws Exce
         return 0;
     }
 
+    /**
+     * A less time-consuming index validation that only looks at the first and last references in the index
+     *
+     * @param reader
+     * @return # of chunks examined, or 0 if there is no browseable index for the reader
+     */
+    public static int lessExhaustivelyTestIndex(final SamReader reader) {
+        // look at all chunk offsets in a linear index to make sure they are valid
+
+        if (reader.indexing().hasBrowseableIndex()) {
+
+            // content is from an existing bai file
+            final CachingBAMFileIndex existingIndex = (CachingBAMFileIndex) reader.indexing().getBrowseableIndex();
+            final int n_ref = existingIndex.getNumberOfReferences();
+
+
+            int chunkCount = 0;
+            int indexCount = 0;
+            for (int i = 0; i < n_ref; i++) {
+
+                final BAMIndexContent content = existingIndex.getQueryResults(i);
+
+                final List<Chunk> chunks = content.getAllChunks();
+                for (int j = 0; j < chunks.size(); j++) {
+
+                    chunkCount++;
+
+                    //validate the first and last chunk in each contig
+                    if (j == 0 || j==(chunks.size()-1)) {

```

Comment:
My understanding is that there could in theory be 1 chunk for every position in the contig.  For large contigs that means many millions of calls to chunks.size() here and on line 111, whereas that could be avoided by making it a final variable above.